craterpy/classes.py

```
@@ -469,3 +475,186 @@ def plot(self, name="", ax=None, alpha=0.2, **kwargs):
+                    data = data.to_crs(target_crs)
+            except Exception as e:
+                raise ValueError(f"Error converting to CRS '{crs}': {str(e)}")
+        
+        # Export to GeoJSON
+        if filename is not None:
+            data.to_file(filename, driver="GeoJSON")
+            return None
+        else:
+            # Return as string
+            return data.to_json()
+
+
+    @classmethod
+    def read_shapefile(cls, filename, body="Moon", units="m"):
+        """
+        Read crater data from a shapefile or GeoJSON file.
+        
+        Parameters
+        ----------
+        filename : str
+            Path to the shapefile or GeoJSON file.
+        body : str, optional
+            Planetary body, e.g. Moon, Vesta (default: Moon).
+            If the file contains a 'body' or 'planet' field, that value will be used instead.
+        units : str, optional
+            Length units of radius/diameter, m or km (default: m).
+            If the file contains a 'units' field, that value will be used instead.
+        
+        Returns
+        -------
+        CraterDatabase
+            A new CraterDatabase instance containing the data from the file.
+            
+        Notes
+        -----
+        This method assumes the file was previously created by CraterDatabase.to_geojson()
+        or has a compatible format with lat/lon coordinates and radius or diameter information.
+        
+        If the file contains different coordinate column names than expected, this method
+        will attempt to identify them by common names (e.g., 'lat', 'latitude', 'lon', 'longitude').
+        """
+        gdf = gpd.read_file(filename)
+        
+        # Check if the file has body or units info
+        if 'body' in gdf.columns and gdf['body'].nunique() == 1:
+            body = gdf['body'].iloc[0]
+        elif 'planet' in gdf.columns and gdf['planet'].nunique() == 1:
+            body = gdf['planet'].iloc[0]
```

I don't think we need to handle body or planet fields:

1. in general users will know what body their shapefile is from so they wouldn't add a "body" or "planet" field when make a file in GIS
2. we're assuming we're reading a file exported by `to_geojson` so if we don't make that field we don't need to handle it
3. we should default to respecting the user-supplied argument rather than what is detected in the file so the user can overwrite data. otherwise their supplied argument silently does nothing 
4. planetary body info is stored in the crs, so if we want to automatically detect the body from a shapefile we should check there instead. 

To handle case 4., I'd make the default arg `body=None` and then `if body is None` check for the planetary body from the crs. If no valid body in crs, raise error and ask them to supply body or crs